src/vmcalgs.inl

```diff
@@ -60,42 +60,43 @@ UIntType MetropolisUpdate_(Wavefunction const &psi, VarParams<V> params, Positio
                            RandomGenerator &gen) {
     static_assert(IsWavefunction<D, N, V, Wavefunction>());
 
-    UIntType succesfulUpdates = 0u;
-    for (Position<D> &p : poss) {
-        Position const oldPos = p;
+    UIntType successfulUpdates;
+    for (Position<D> p : poss) {
+        Position<D> const oldPos = p;
         FPType const oldPsi = psi(poss, params);
-        std::uniform_real_distribution<FPType> unifDist(0, 1);
+        // Using float avoids narrowing, whatever FPType is
+        std::uniform_real_distribution<FPType> unifDist(-0.5f, 0.5f);
         std::transform(p.begin(), p.end(), p.begin(), [&gen, &unifDist, step](Coordinate c) {
-            return Coordinate{c.val + (unifDist(gen) - FPType{0.5f}) * step};
+            return Coordinate{c.val + unifDist(gen) * step};
         });
         if (unifDist(gen) > std::pow(psi(poss, params) / oldPsi, 2)) {
             p = oldPos;
 
         } else {
-            ++succesfulUpdates;
+            ++successfulUpdates;
         }
     }
-    return succesfulUpdates;
+    return successfulUpdates;
 }
 
 // Should be hidden from the user
-// Computes the local energy with the analytic formula for the derivative of the wavefunction
-template <Dimension D, ParticNum N, VarParNum V, class Wavefunction, class KinEnergy, class Potential>
-Energy LocalEnergyAnalytic_(Wavefunction const &psi, VarParams<V> params, KinEnergy const &kin,
-                            Potential const &pot, Positions<D, N> poss) {
+// Computes the drift force using its analytic expression
+template <Dimension D, ParticNum N, VarParNum V, class Wavefunction>
+void DriftForceAnalytic_(Wavefunction const &psi, VarParams<V> params, Positions<D, N> poss,
+                         Position<D> &driftForce) {
     static_assert(IsWavefunction<D, N, V, Wavefunction>());
-    static_assert(IsKinEnergy<D, N, V, KinEnergy>());
-    static_assert(IsPotential<D, N, Potential>());
 
-    return Energy{kin(poss, params) / psi(poss, params) + pot(poss)};
+    FPType alpha = params[0];
+    for (int i = 0; i < D; ++i) {
+        driftForce[i].val = -4 * alpha * poss[i].val;
```

Comment:
You are assuming that the wavefunction is the one of the harmonic oscillator!!! :(
That defeats the point of using templates, since this function only works in the harmonic oscillator case!
Ask for the analytical expression of the first derivative and update accordingly.